src/main.rs

```diff
@@ -49,7 +49,15 @@ fn main() -> io::Result<()> {
 
     let bin_name = prog.value_of("binary").unwrap();
 
-    let scale: usize = prog.value_of("scale").unwrap().parse().unwrap();
+    let scale: usize = match prog.value_of("scale")
+                            .unwrap()
```

Comment:
move up to `unwrap()` to a single line, like `bin_name`. Split clap-rs and argument parsing.
```rust
let scale = prog.value_of("scale").unwrap();
let scale: usize = match prog.parse()
// ...

```